### PR TITLE
Add path-prefix functionality

### PR DESCRIPTION
**Pull request by tipsy**

### CODE REVIEW

**Comment by tipsy** [comment]
Usage:
```
path("/api", () -> {
    post("/add-role",        Api.addRole);
    post("/add-email",       Api.addEmail);
    delete("/remove-email",  Api.addMsisdn);
});
```

**Comment by tipsy** [comment]
Any thoughts on this @perwendel? @jakaarl? @OpsRJ?

**Comment by jakaarl** [comment]
I love the usage, very fluent.

File: src/main/java/spark/Service.java [hunk 1]
```diff
@@ -71,6 +73,7 @@
     protected Optional<Integer> webSocketIdleTimeoutMillis = Optional.empty();
 
     protected EmbeddedServer server;
+    protected Stack<String> pathStack = new Stack<>();
```
Comment (jakaarl) [inline src/main/java/spark/Service.java]: I think `Stack` is one of those "not recommended" classes, quoting:
> A more complete and consistent set of LIFO stack operations is provided by the Deque interface and its implementations, which should be used in preference to this class. 

Stack is, IIRC, synchronized and eg. `ArrayDeque` probably not, but from what I understood (took a quick glance only, though) there is no need for it here. And if there was, using `Stack` wouldn't really close the window of opportunity for a race condition.

**Comment by tipsy** [comment]
Your comment vanished when I amended the commit, sorry about that. I have change to Deque now, as well as added some more tests (filters, triply nested paths) and documentation.

**Comment by MouettE-SC** [comment]
👍 like it

**Comment by OpsRJ** [comment]
@tipsy Very nice. Loved it. Thanks!

**Comment by perwendel** [comment]
Good stuff!

**Pull request merged**